### DISCUSSION TITLE
Separation of the local cache for release/snapshot artifacts

### DISCUSSION THREAD

**Discussion by khmarbaise** [support-and-care/maven-support-and-care#31] [Ideas for Funding]
The idea to make a separation between release versions and `SNAPSHOT` versions. This means having a location for release artifacts which will not change very often (except if you change a pom file) and the `SNAPSHOT` which are usually only put into the local cache on intention via `mvn ... install`. This would make it easier to handle container image builds.

There could also the idea to separate the artifacts separated by the remote repository from which it is downloaded (not sure about this)?

**Comment by rmannibucau** [comment]
Think we already have it but luckily it is not on by default since a single repo is key for most consumers.

**Comment by slawekjaranowski** [comment]
There is a `Split Local Repository` - https://maven.apache.org/resolver/local-repository.html

As I remember IntelliJ doesn't understand a split repository

**Comment by gnodet** [comment]
Yes, this is already implemented. The problem is if/how we can make is the default behaviour.

**Comment by cstamas** [comment]
My related ramblings about this problem: https://maveniverse.eu/blog/2025/11/09/maven-local-repository-ii/

Some ideas:
* remotely cached artifacts _are immutable_, we should make use of this fact
* locally built and installed artifacts _are mutable_
* split repo seems too invasive, but we may be getting there; but not as long as Maven 3 needs to be supported (Resolver 1 compatibility)
* having a global cache (growing, containing immutable artifacts) may be the key here; hardlinking like Mimir
* installing maybe needs to happen "per workspace" or "per project" or alike (idea: not always globally)
